cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java

```diff
@@ -344,15 +344,19 @@ public static Location getParent(Location location) {
   /**
    * Given a locationPath which is derived from URI.getPath() and a locationFactory
    * this method generates a Location with URI generated from the given Absolute path
+   * The given absolutePath is expected to contain basePath if any.
```

Comment:
this entire javadoc is confusing and should change. Just say something like 'Get a Location using a specified location factory and absolute path.'